### PR TITLE
[xcm-emulator] Decouple the `AccountId` type from `AccountId32`

### PR DESCRIPTION
**Pull request by NachoPal**
Closes: #1381 

Originally from: https://github.com/paritytech/cumulus/pull/3037

### CODE REVIEW

File: cumulus/xcm/xcm-emulator/src/lib.rs [hunk 1]
```diff
@@ -1326,9 +1329,9 @@ pub struct TestArgs {
 }
 
 /// Auxiliar struct to help creating a new `Test` instance
-pub struct TestContext<T> {
-	pub sender: AccountId,
-	pub receiver: AccountId,
+pub struct TestContext<T, R: Chain, D: Chain> {
```
Comment (bkontur) [inline cumulus/xcm/xcm-emulator/src/lib.rs:1332]: what about aligning this R/D chain types?

this is little bit confusing:
sender -> **R** -> Origin + `Origin::RuntimeOrigin` looks not nice
**r**eceiver -> D -> Destination

I would suggest

`R: Chain` -> `SourceChain: Chain`
`D: Chain` -> `TargetChain: Chain`

and bellow:
Origin -> `SourceChain`
Destination -> `TargetChain`

**Review by bkontur — APPROVED** [review]
lgtm, just one nit with type renaming

File: cumulus/xcm/xcm-emulator/src/lib.rs (hunk 1 above)
Comment (NachoPal) [reply cumulus/xcm/xcm-emulator/src/lib.rs]: Renamed `R` and `D` to be aligned with the rest of the generics. I'll keep `Origin` and `Destination` names though as they are also descriptive.

Comment (bkontur) [reply cumulus/xcm/xcm-emulator/src/lib.rs]: @NachoPal
ok, no problem :)

hrmp uses similar sender/recipient notation [here](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/parachain/src/primitives.rs#L300-L304), but I guess receiver is also ok

**Review by joepetrowski — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Review by gilescope — APPROVED** [review]

**Pull request merged**